### PR TITLE
GHA: Several fixes on depexts workflow

### PR DESCRIPTION
**Pull request by rjbou**
Major ones:
* Fix centos by faking on with almalinux
* Fix conf-* packages install and check
* Add packages with `os-version` to check
* More caching in docker build

_I'll backport once green_

### CODE REVIEW

File: .github/scripts/depexts/generate-actions.sh [hunk 1]
```diff
@@ -42,9 +42,12 @@ EOF
...
    OCAML_CONSTRAINT=' & < "5.0"'
     cat >$dir/Dockerfile << EOF
-FROM centos:7
+FROM almalinux:9.4
+RUN dnf install 'dnf-command(config-manager)' -y
+RUN dnf config-manager --set-enabled crb
 RUN yum install -y $mainlibs $ocaml
-RUN yum install -y gcc-c++
+RUN yum install -y gcc-c++ diffutils
+RUN sed -i 's/ID="almalinux"/ID="centos"/' /etc/os-release
```
Comment (kit-ty-kate) [inline .github/scripts/depexts/generate-actions.sh:50]: i don't think we should try to emulate centos

Comment (rjbou) [reply .github/scripts/depexts/generate-actions.sh:50]: why not?

Comment (kit-ty-kate) [reply .github/scripts/depexts/generate-actions.sh:50]: centos is effectively a dead platform as far as the community goes. Trying to emulate a distribution that is diverging is a bad idea

Comment (rjbou) [reply .github/scripts/depexts/generate-actions.sh:50]: It is a dead platform true, but there are still users on it, and it is still supported by opam.
Almalinux is meant to be a replacement to centos, providing backward compatibility. Until it diverges completely, it is a good testing platform for centos jobs.
We could add a full support for almalinux & rockylinux, but that's another issue, and they should have their own jobs.

**Review by kit-ty-kate — APPROVED** [review]

**Pull request merged**